src/sentry/integrations/issues.py

```diff
@@ -210,7 +210,9 @@ def get_repository_choices(self, group, **kwargs):
 
         params = kwargs.get('params', {})
         defaults = self.get_project_defaults(group.project_id)
-        default_repo = params.get('repo', defaults.get('repo') or repo_choices[0][0])
+        # Both Azure DevOps and GitLab have projects instead of repos
+        field_name = kwargs.get('field_name', 'repo')
```

Comment:
Hi @markstory, so I'm having a bit of trouble understanding how passing `repo` in this method is an improvement over allowing the subclass to specify a `field_name` in `kwargs`. In this case, the only difference is that the name of field is `'project'` instead of `repo`.

Reply:
Having `field_name` creates a wider coupling between the subclass and super-class. To me, having a super-class handle variation that exists in the sub-classes implies that the abstraction provided by the super-class is leaky or incomplete.

Given that we control all aspects of these implementations normalizing on one field/term cuts down on the number of terms/language we have to remember and reduce the amount of indirection between different layers in the class tree. If we were mapping between vendor terminology/data structures and our own I think this would totally make sense.

I don't think this is a big deal either way as integrations are a small part of sentry. If this is extended into a generalized pattern, it will adds incremental complexity where it could likely be avoided.